lib/manager/kustomize/extract.ts

```diff
@@ -18,49 +18,32 @@ interface Kustomize {
   images: Image[];
 }
 
-// extract the version from the url
-const versionMatch = /(?<basename>.*)\?ref=(?<version>.*)\s*$/;
-
-// extract the url from the base of a url with a subdir
-const extractUrl = /^(?<url>.*)(?:\/\/.*)$/;
-
-const githubUrl = /^github\.com\/(?<depName>(?<lookupName>[^/]+?\/[^/]+?)(?:\/[^/]+?)*)\?ref=(?<currentValue>.+)$/;
+// URL specifications should follow the hashicorp URL format
+// https://github.com/hashicorp/go-getter#url-format
+const gitUrl = /^(?:git::)?(?<url>((?:http|https|ssh):\/\/(?:.*@)?)?(?<path>([^:/]+[:/])?(?<project>[^/]+\/[^/]+)))(?<subdir>[^?]*)\?ref=(?<currentValue>.+)$/;
 
 export function extractBase(base: string): PackageDependency | null {
-  const githubMatch = githubUrl.exec(base);
-
-  if (githubMatch?.groups) {
-    const { currentValue, depName, lookupName } = githubMatch.groups;
+  const match = gitUrl.exec(base);
 
-    return {
-      datasource: datasourceGitHubTags.id,
-      depName,
-      lookupName,
-      currentValue,
-    };
+  if (!match) {
+    return null;
   }
 
-  const basenameVersion = versionMatch.exec(base);
-  if (basenameVersion) {
-    const currentValue = basenameVersion.groups.version;
-    const root = basenameVersion.groups.basename;
-
-    const urlResult = extractUrl.exec(root);
-    let url = root;
-    // if a match, then there was a subdir, update
-    if (urlResult && !url.startsWith('http')) {
-      url = urlResult.groups.url;
-    }
-
+  if (match?.groups.path.includes('github.com')) {
```

Comment:
```suggestion
  if (match?.groups.path.startsWith('github.com')) {
```

This should fix the CodeQL alert